torch/csrc/jit/codegen/cuda/index_compute.cpp

```diff
@@ -2166,7 +2166,8 @@ std::vector<Val*> Index::getProducerStridedIndices(
   }
 
   std::vector<Val*> strided_indices;
-  if (producer->getMemoryType() == MemoryType::Global) {
+  if (producer->getMemoryType() == MemoryType::Global ||
+      ir_utils::isRandOp(consumer->definition())) {
```

Comment:
This doesn't seem right to me. Are you just using this as a convenient mechanism to get the index for the RNG offset?

Reply:
Can you make a direct interface to do this? We should just call consumer global indexing with full contiguity. We should just be able to make a small modification to `ir_utils::TVDomainGuard` to modify the contiguity of the consumer to grab this index.

Reply:
Yes, would be helpful to separate out an RNG index interface that uses the consumer tv only. Also helps with further rng layout optimizations.